frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostsPage.tsx

```diff
@@ -52,10 +52,33 @@ type BareMetalHostsPageProps = {
   hasNodeMaintenanceCapability: boolean;
 };
 
+const getCreateProps = ({ namespace }: { namespace: string }) => {
+  const items: any = {
```

Comment:
I don't see this being used anywhere else except kubevirt plugin. The standard used everywhere in console is to use link to the form with an option to switch to yaml editor afterwards.

I am inclined to using that as It makes more sense to me from the UX point of view - GUI provides a UI based form by default and lets user switch to yaml editor if needed. There's no need for the user to make the initial decision of what creation dialog type to use.

Reply:
> I don't see this being used anywhere else except kubevirt plugin. The standard used everywhere in console is to use link to the form with an option to switch to yaml editor afterwards.

It is used for example in `Secrets`

> 
> I am inclined to using that as It makes more sense to me from the UX point of view - GUI provides a UI based form by default and lets user switch to yaml editor if needed. There's no need for the user to make the initial decision of what creation dialog type to use.

It can make the user falsely think it is just a mode switch between two views (form and modal) and this can lead to discarding the filled values. 

IMO it is better to give the user an option to choose what type of a dialog to use, instead of making additional redirect.

Reply:
> It can make the user falsely think it is just a mode switch between two views (form and modal) and this can lead to discarding the filled values.

OK, makes sense.
We should try to fix this UX problem across console. Either by using the dropdown in all applicable places or ultimately make the yaml link the actual mode switch, which is quite non trivial as certain resource creation operates on multiple yaml files (e.g. Adding BM Host creates Secret and BM Host resource.)

Reply:
the mode switch has many other problems - wizards, different approaches (import vs create VM) and already mentioned multiple resources